### PR TITLE
Fixes provisioning on subnet with static boot mode

### PR DESCRIPTION
**Pull request by ares**
@stbenjam could you test please?

### CODE REVIEW

File: snippets/kickstart_networking_setup.erb [hunk 1]
```diff
@@ -19,6 +19,9 @@ BOOTPROTO="<%= dhcp ? 'dhcp' : 'none' -%>"
 <% unless dhcp -%>
 IPADDR="<%= @host.ip -%>"
 NETMASK="<%= subnet.mask -%>"
+<% if subnet.gateway.present? %>
```
Comment (stbenjam) [inline snippets/kickstart_networking_setup.erb]: You can't do present? in the jail

Comment (stbenjam) [inline snippets/kickstart_networking_setup.erb]: `There was an error rendering the Katello Kickstart Default template: The snippet 'kickstart_networking_setup' threw an error: undefined method 'present?' for String::Jail (String)`

Comment (stbenjam) [inline snippets/kickstart_networking_setup.erb]: But .empty? is allowed...

File: snippets/kickstart_networking_setup.erb [hunk 2]
```diff
@@ -19,6 +19,9 @@ BOOTPROTO="<%= dhcp ? 'dhcp' : 'none' -%>"
 <% unless dhcp -%>
 IPADDR="<%= @host.ip -%>"
 NETMASK="<%= subnet.mask -%>"
+<% if !subnet.gateway.nil? && !subnet.gateway.empty? %>
```
Comment (stbenjam) [inline snippets/kickstart_networking_setup.erb]: `-%>` please

**Comment by stbenjam** [comment]
Works for me :+1:  Just one small nitpick

**Comment by domcleal** [comment]
Just to check, does this still work on EL6?  Often GATEWAY gets added into /etc/sysconfig/network rather than one of the interfaces, and I'm wondering if that's added automatically on EL6 or not...

**Comment by stbenjam** [comment]
@domcleal That's left up to anaconda, the snippet doesn't touch /etc/sysconfig/network so it will do the right thing AFAICS - I don't have easy access to provision an el6 machine quickly though.

**Comment by domcleal** [comment]
Could GATEWAY be set twice then, once in /etc/sysconfig/network by anaconda and again by this snippet?  (Assuming anaconda does indeed do that.)

**Comment by stbenjam** [comment]
Yes, but I don't think it's a problem, a quick check on my el6 boxes seems to be that the interface-level gateway takes precedence over /etc/sysconfig/network.

**Comment by domcleal** [comment]
Right, that makes sense.  Thanks!

**Comment by ares** [comment]
@stbenjam updated, I guess you want me to prepare the same patch for katello

**Comment by stbenjam** [comment]
@ares We pull in the foreman snippet, we just need this merged and synced with foreman develop, and I can merge your Katello PR's.

**Comment by ares** [comment]
ah, right, this one is not changed in katello, anyway I created a Foreman issue http://projects.theforeman.org/issues/9510 for this so it's tracked

**Comment by domcleal** [comment]
Thanks @ares!

**Pull request merged**